coreax/kernel.py

```diff
@@ -102,6 +102,55 @@ def median_heuristic(x: ArrayLike) -> Array:
 class Kernel(eqx.Module):
     """Abstract base class for kernels."""
 
+    def __add__(self, addition: Kernel | int | float):
+        """Overload `+` operator."""
+        if not isinstance(addition, (Kernel, int, float)):
```

Comment:
It would be more idiomatic if these were if, elif, else return NotImplemented. Type hints are missing on the return. Can we also use the Union instead of `|`

Reply:
Yep much nicer, done